apps/site/hooks/react-client/useDetectOS.ts

```diff
@@ -15,6 +15,37 @@ type UserOSState = {
   architecture: Architecture | '';
 };
 
+// taken from https://github.com/faisalman/ua-parser-js/issues/732#issue-2348848266
+function isAppleSilicon() {
+  try {
+    // Best guess if the device uses Apple Silicon: https://stackoverflow.com/a/65412357
+    const webglContext = document.createElement('canvas').getContext('webgl');
+    if (webglContext == null) {
+      return false;
+    }
+    const debugInfo = webglContext.getExtension('WEBGL_debug_renderer_info');
+    const renderer =
+      (debugInfo &&
+        webglContext.getParameter(debugInfo.UNMASKED_RENDERER_WEBGL)) ||
+      '';
+    if (renderer.match(/Apple/) && !renderer.match(/Apple GPU/)) {
+      return true;
+    }
+
+    if (
+      webglContext
+        .getSupportedExtensions()
+        ?.includes('WEBGL_compressed_texture_s3tc_srgb')
+    ) {
+      return true;
+    }
+  } catch {
+    /**/
+  }
+
+  return false;
+}
+
```

Comment:
Also, this approach isn't [the one the authors of that library ended up going with](https://github.com/faisalman/ua-parser-js/commit/c391d8a73c1a030c24845bc191b021bfb6cba256)


```suggestion
// Ref: https://github.com/faisalman/ua-parser-js/commit/c391d8a73c1a030c24845bc191b021bfb6cba256
function isAppleSilicon() {
    try {
        const canvas = document.createElement('canvas');
        const webgl = canvas.getContext('webgl2') || canvas.getContext('webgl') || canvas.getContext('experimental-webgl');
        const debug = webgl.getExtension('WEBGL_debug_renderer_info');
        const renderer = webgl.getParameter(debug.UNMASKED_RENDERER_WEBGL);
        if (renderer.match(/apple m\d/i)) {
            return true;
        }
    } catch {
        return false;
    }
}

```

Reply:
sure, i can pull in that approach if it's better.

Reply:
more importantly, ua-parser-js is [AGPL, and as far as I know the strong copy-left](https://github.com/faisalman/ua-parser-js?tab=AGPL-3.0-1-ov-file) would prohibit MIT usage

Reply:
When that is the only issue left, I'll reach out and see if I can get written permission to consider the snippet MIT.